torch/quantization/custom_module_class_mappings.py

```diff
@@ -0,0 +1,75 @@
+OBSERVED_CUSTOM_MODULE_CLASS_MAPPINGS = dict()
+
+def register_observed_custom_module_mapping(custom_module_class, observed_custom_module_class):
+    """ Register a mapping from `custom_module_class` to
+    `observed_custom_module_class`
+    `observed_custom_module_class` will have a `from_float` classmethod,
+    which will return an observed custom module instance given
+    a float custom module instance.
+    This will be used in prepare step of post training static quantization or
+    quantization aware training
+    """
+    assert hasattr(observed_custom_module_class, 'from_float'), 'from_float must be' + \
+        ' defined in observed custom module class'
+    OBSERVED_CUSTOM_MODULE_CLASS_MAPPINGS[custom_module_class] = \
+        observed_custom_module_class
+
+def get_observed_custom_module_class(custom_module_class):
+    """ Get the corresponding observed module class for a given
+    custom module.
+    """
+    observed_custom_module_class = \
+        OBSERVED_CUSTOM_MODULE_CLASS_MAPPINGS.get(custom_module_class, None)
+    assert observed_custom_module_class is not None, \
+        'Custom module class {}'.format(custom_module_class) + \
+        ' does not have a corresponding observed module class'
+    return observed_custom_module_class
+
+QUANTIZED_CUSTOM_MODULE_CLASS_MAPPINGS = dict()
+
+def register_quantized_custom_module_mapping(custom_module_class, quantized_custom_module_class):
```

Comment:
should this is `observed_custom_module_class`? otherwise it's unclear if the input should be observed, not observed, or either

Reply:
it's not observed, will change this to float_custom_module_class